shared/prow/prow.go

```diff
@@ -0,0 +1,336 @@
+type Build struct {
+	JobName	    string
+	JobType     string
+	BuildID	    int
+	Bucket      string // optional
+}
+
+// Started holds the started.json values of the build.
+type Started struct {
+	Timestamp   int64             `json:"timestamp"` // epoch seconds
+	RepoVersion string            `json:"repo-version"`
+	Node        string            `json:"node"`
+	Pull        string            `json:"pull"`
+	Repos       map[string]string `json:"repos"` // {repo: branch_or_pull} map
+}
+
+// Finished holds the finished.json values of the build
+type Finished struct {
+	// Timestamp is epoch seconds
+	Timestamp   int64            `json:"timestamp"`
+	Passed      bool             `json:"passed"`
+	JobVersion  string           `json:"job-version"`
+	Metadata    Metadata         `json:"metadata"`
+	running     bool
+}
+
+// Metadata contains metadata in finished.json
+type Metadata map[string]interface{}
+
+/* Local logics */
+
+// GetLocalArtifactsDir gets the aritfacts directory where prow looks for artifacts.
+// By default, it will look at the env var ARTIFACTS.
+func GetLocalArtifactsDir() string {
+	dir := os.Getenv("ARTIFACTS")
+	if dir == "" {
+		log.Printf("Env variable ARTIFACTS not set. Using %s instead.", ArtifactsDir)
+		dir = ArtifactsDir
+	}
+	return dir
+}
+
+
+/* GCS related logics */
+
+// Initialize wraps gcs authentication, have to be invoked before any other functions
+func Initialize(sa string) error {
+	return gcs.Authenticate(ctx, sa)
+}
+
+// NewJob creates new job struct
+func NewJob(jobName string) *Job {
+	return &Job{
+		Name: jobName,
+		Bucket: BucketName,
+	}
+}
+
+// NewBuild creates new build struct
+func NewBuild(jobName string, buildID int) *Build {
+	return &Build{
+		Bucket: BucketName,
+		JobName: jobName,
+		BuildID: buildID,
+	}
+}
+
+// GetPath gets gcs path of current job
+func (j *Job) GetPath() string {
+	if nil != j.Prefix { // return custom path if set
+		return *j.Prefix
+	}
+
+	var res string
+	switch j.Type {
+		case PeriodicJob, PostsubmitJob:
+			res = path.Join("logs", j.Name)
+		case PresubmitJob:
+			res = path.Join("pr-logs", "pull", OrgName + "-" + j.Repo, j.Commit, j.Name)
+		case BatchJob:
+			res = path.Join("pr-logs", "pull", "batch", j.Name)
+		default:
+			log.Fatalf("unknown job spec type: %v", j.Type)
+	}
+
+	// cache prefix if computed
+	j.Prefix = &res
+	return res
+}
+
+// GetLatestBuildNumber gets the latest build number for job
+func (j *Job) GetLatestBuildNumber() (int, error) {
+	logFilePath := path.Join(j.GetPath(), Latest)
+	log.Printf("Using %s to get latest build number", logFilePath)
+	contents, err := gcs.Read(ctx, BucketName, logFilePath)
+	if err != nil {
+		return 0, err
+	}
+	latestBuild, err := strconv.Atoi(strings.TrimSuffix(string(contents), "\n"))
+	if err != nil {
+		return 0, err
+	}
+
+	return latestBuild, nil
+}
+
+// GetBuild gets build struct based on job info
+// 	No gcs operation
+func (j *Job) GetBuild(buildID int) *Build {
```

Comment:
Since you're not fetching anything, what about calling this "NewBuild"?

Reply:
renamed